src/main/java/com/leinardi/pycharm/mypy/util/Async.java

```diff
@@ -47,7 +48,12 @@ private Async() {
 
     public static <T> Future<T> whenFinished(final Future<T> future) {
         while (!future.isDone() && !future.isCancelled()) {
-            ProgressManager.checkCanceled();
+            try {
+                ProgressManager.checkCanceled();
+            } catch (ProcessCanceledException e) {
```

Comment:
I think you should catch any `RuntimeException`. If anything gets thrown, the future result is no longer used.

Reply:
Oh I see `checkCanceled` is explicit that it will only throw `ProcessCanceledException`. Never mind then :)